src/qt/guiutil.cpp

```diff
@@ -374,6 +374,12 @@ void bringToFront(QWidget* w)
         // activateWindow() (sometimes) helps with keyboard focus on Windows
         if (w->isMinimized()) {
             w->showNormal();
+            // This is a workaround when the main window still minimized to the taskbar (iconified)
+            // due to a (possible) Qt bug or a weird window manager behaviour.
+            if (isObscured(w)) {
```

Comment:
Does calling `isObscured()` directly after `showNormal()` make sense or should it hit back the the runloop first?